core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -259,15 +260,16 @@ class ControllerBrokerRequestBatch(controller: KafkaController) extends  Logging
     if (stopReplicaRequestMap.nonEmpty)
       throw new IllegalStateException("Controller to broker state change requests batch is not empty while creating a " +
         "new one. Some StopReplica state changes %s might be lost ".format(stopReplicaRequestMap.toString()))
-    if (updateMetadataRequestMap.nonEmpty)
+    if (updateMetadataRequestBrokerSet.nonEmpty)
       throw new IllegalStateException("Controller to broker state change requests batch is not empty while creating a " +
-        "new one. Some UpdateMetadata state changes %s might be lost ".format(updateMetadataRequestMap.toString()))
+        "new one. Some UpdateMetadata state changes to brokers %s might be lost ".format(updateMetadataRequestBrokerSet.toString()))
```

Comment:
I think we used to see this exception got thrown before. In general the messages in the exception should provide us information for debug. In this case, it may be worth knowing what partition information to which broker set were still pending. So we can trace back and match the logs from the state machine, and further discover the potential triggering event.